pkg/services/publicdashboards/database/database.go

```diff
@@ -267,3 +268,20 @@ func (d *PublicDashboardStoreImpl) Delete(ctx context.Context, orgId int64, uid
 
 	return affectedRows, err
 }
+
+func (d *PublicDashboardStoreImpl) GetByDashboardFolder(ctx context.Context, dashboard *dashboards.Dashboard) ([]*PublicDashboard, error) {
+	if dashboard == nil || !dashboard.IsFolder {
+		return []*PublicDashboard{}, nil
+	}
+
+	var pubdashes []*PublicDashboard
+
+	err := d.sqlStore.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		return sess.SQL("SELECT * from dashboard_public WHERE dashboard_uid IN (SELECT uid FROM dashboard WHERE folder_id = ?)", dashboard.ID).Find(&pubdashes)
```

Comment:
The `dashboard_uid` is not enough for dashboard identification, you need to use the `org_id` as well.

Reply:
good call 👍 Will fix